tests/test_composite.py

```diff
@@ -324,18 +324,35 @@ def test_multi_slice_index(ant, sphere, uniform, airplane, globe):
         assert id(sub[i]) == id(multi[i])
         assert sub.get_block_name(i) == multi.get_block_name(i)
     sub = multi[0:-1]
-    assert len(sub) == len(multi) == multi.n_blocks
-    for i in range(multi.n_blocks):
+    assert len(sub) + 1 == len(multi) == multi.n_blocks
```

Comment:
It turns out that slicing was broken more than I thought.

Reply:
I used `len` throughout this test.  Checking that `len` and `n_blocks` return the same here would be misplaced.